gql-sql-slicer.ts

```diff
@@ -39,6 +39,156 @@ interface metricDataResolver {
+        join(arg.name.value)(arg.value, query, knex)
+        return res
+      } else {
+        throw "Join can't be called inside of join"
+      }
+    }
+
+    if (arg.name.value.endsWith('_gt'))
+      return res.concat([
+        [
+          buildFullName(args, query, arg.name.value.replace('_gt', ''), false),
+          '>',
+          arg.value.value,
+        ],
+      ])
+    if (arg.name.value.endsWith('_gte'))
+      return res.concat([
+        [
+          buildFullName(args, query, arg.name.value.replace('_gte', ''), false),
+          '>=',
+          arg.value.value,
+        ],
+      ])
+    if (arg.name.value.endsWith('_lt'))
+      return res.concat([
+        [
+          buildFullName(args, query, arg.name.value.replace('_lt', ''), false),
+          '<',
+          arg.value.value,
+        ],
+      ])
+    if (arg.name.value.endsWith('_lte'))
+      return res.concat([
+        [
+          buildFullName(args, query, arg.name.value.replace('_lte', ''), false),
+          '<=',
+          arg.value.value,
+        ],
+      ])
+    if (arg.name.value.endsWith('_like'))
+      return res.concat([
+        [
+          buildFullName(
+            args,
+            query,
+            arg.name.value.replace('_like', ''),
+            false,
+          ),
+          'LIKE',
+          arg.value.value,
+        ],
+      ])
+    if (arg.name.value.endsWith('_in'))
+      return res.concat([
+        [
+          buildFullName(args, query, arg.name.value.replace('_in', ''), false),
+          'in',
+          arg.value.value.split('|'),
+        ],
+      ])
+    return res.concat([
+      [buildFullName(args, query, arg.name.value, false), '=', arg.value.value],
+    ])
+  }, [])
+}
+
+function buildFullName(
+  args: any | any[],
+  query,
+  field: string,
+  evaluateOnlyWithLinkSymbol = true,
+) {
+  args = Array.isArray(args) ? argumentsToObject(args) : args
+  const table = args.from || query.table
+
+  if (!field.startsWith('@') && (evaluateOnlyWithLinkSymbol || !args.from)) {
+    return field
+  } else {
+    return `${table}.${field.replace(/^@/, '')}`
+  }
+}
+
+function join(type: JoinType) {
+  return (tree, query, knex) => {
+    if (!tree.arguments && !tree.fields)
+      throw 'Join function requires arguments'
+
+    const args = argumentsToObject(tree.arguments || tree.fields)
+    if (!args.table) throw "Join function requires 'table' as argument"
+
+    const byKeys = [
+      'by',
+      'by_gt',
+      'by_gte',
+      'by_lt',
+      'by_lte',
+      'by_like',
+      'by_in',
+    ].filter((key) => args[key] !== undefined)
+
+    if (!byKeys.length) throw "Join function requires 'by' as argument"
+
+    const filters = transformFilters(
+      (tree.arguments || tree.fields)
+        .filter(({ name: { value } }) => byKeys.includes(value))
+        .concat({ name: { value: 'from' }, value: { value: args.table } }),
+      query,
+    )
+
+    query.promise[type](args.table, function () {
```

Comment:
Not sure I understand - query.promise is a promise chain, right, how come it has properties now?